filecoin-virtual-machine/section-a.md

```diff
@@ -0,0 +1,307 @@
+
+The time between upload and being able to get your PoDSI should only be a few minutes. You can get the PoDSI for your file by calling the `getProof` function in one of the following ways:
+
+```bash
+# Assumes that uploaded your file to mainnet.
+# Alternatively, if you are using testnet, add &network=testnet to the end of the URL.
+curl https://api.lighthouse.storage/api/lighthouse/get_proof?cid=<puppy_CID>
+```
+
+```javascript
+let response = await axios.get("https://api.lighthouse.storage/api/lighthouse/get_proof", {
+    params: {
+        cid: lighthouse_cid,
+        network: "testnet" // Change the network to mainnet when ready
+    }
+})
+```
+
+### Get your deal ID from your upload
+When you upload the picture of your puppy, the on-chain deal that is made to store it on the Filecoin network is assigned a unique deal ID. You can get this deal ID the same way you get the PoDSI for your file. In the above, it would be accessible through `response.data.deal_id`.
+
+Under the hood, the node infrastructure is working hard to ensure that your file is included on-chain. The process can take up to about **an hour**.
+
+### Download your file using the file’s CID
+Now that your file is stored on the Filecoin network, you can retrieve it using the file's CID. You can do this by calling the `download` function in one of the following ways:
+
+```bash
+# Assumes that you have lighthouse-cli installed. If not, feel free to download it using 
+# npm install -g @lighthouse-web3/sdk
+lighthouse-web3 decrypt-file <CID>
+```
+
+```javascript
+let response = await axios({
+    method: 'GET',
+    url: `${lighthouseDealDownloadEndpoint}${lighthouse_cid}`,
+    responseType: 'stream',
+});
+
+try {
+    const filePath = await this.saveResponseToFile(response, downloadPath);
+    console.log(`File saved at ${filePath}`);
+    return filePath
+} catch (err) {
+    console.error(`Error saving file: ${err}`);
+}
+
+saveResponseToFile(response, filePath) {
+    const writer = fs.createWriteStream(filePath);
+
+    // Pipe the response data to the file
+    response.data.pipe(writer);
+
+    return new Promise((resolve, reject) => {
+        writer.on('finish', () => resolve(filePath));
+        writer.on('error', (err) => {
+            console.error(err);
+            reject(err);
+        });
+    });
+}
+```
+
+## Why does all this matter?
+
+We see a bright future in enabling permanent, immutable, decentralized data-storage for developers. 
+
+The interface for the lighthouse SDK is designed to be simple and easy to use. We hope that this will enable developers to easily integrate the Filecoin network as the primary data storage provider for their applications.
+
+More importantly, this enables developers to build novel applications. Imagine a dapp or DAO that can be built to incentivize, analyze and store upload metadata on-chain. There are a couple of examples of this:
+
+- Rewarding $TOKEN based on the upload of a particular file and their CID.
```

Comment:
It would be good to see some Solidity pseudocode describing a flow that rewarded a user with $TOKEN for an upload. Work with Kevin, Sarah and Lighthouse to see what is possible here.

Reply:
Added